Quick Overview of ChannelIDs, Balances and Aliases

blckbx opened the pull request:
PR: 
Adding a quick overview of active channels containing basic information
"channel id | local balance|remote balance | peer alias"

ChanID: xxxxxxxxxxxxxx |  x,xxx,xxx| x,xxx,xxx | Alias: xxx

for quick manual and distinct rebalance actions. This (for me) comes in handy gathering information fast without scrolling channel lists or switching to graphical interfaces to verify chanids, balances and aliases.

**Usage:** rebalance.py -la

This is my first PR and forked project in github, so pls excuse any mistakes I made in the process. 
Pls feel free to review.

C-Otto commented on rebalance.py:
```diff
@@ -176,13 +176,28 @@ def show_channel(self, channel, reverse=False):
         print(get_capacity_and_ratio_bar(channel, self.lnd.get_max_channel_capacity()))
         print("")
 
+    #addition list all chanids
```
Please remove the comment

C-Otto commented on rebalance.py:
```diff
@@ -176,13 +176,28 @@ def show_channel(self, channel, reverse=False):
         print("")
 
+    #addition list all chanids
+    def list_all_chanids(self, reverse=False):
```
Maybe call this "list_channels_compact"? It shows more than just the channel IDs, so the name isn't very fitting.

C-Otto replied:
Please remove the `reverse` parameter, it's always `False`.

C-Otto commented on rebalance.py:
```diff
@@ -176,13 +176,28 @@ def show_channel(self, channel, reverse=False):
             self.show_channel(channel)
             sys.exit(0)
 
+        #addition list all chanids
```
Please remove the comment

C-Otto commented on rebalance.py:
```diff
@@ -280,7 +295,9 @@ def main():
     last_hop_channel_id = arguments.to
-    if not arguments.list_candidates and last_hop_channel_id is None and first_hop_channel_id is None:
+
+    #addition list all chanids
```
Please remove the comment

C-Otto commented on rebalance.py:
```diff
@@ -301,6 +318,14 @@ def get_argument_parser():
         dest="grpc",
         help="(default localhost:10009) lnd gRPC endpoint",
     )
+    #addition list all chanids
```
Please remove the comment

C-Otto commented on rebalance.py:
```diff
@@ -301,6 +318,14 @@ def get_argument_parser():
+        "--list-chanids",
+        action="store_true",
+        dest="listallchanids",
+        help="list all active candidates' chanids",
```
That's not a fitting description. "Shows a compact list of all channels, one per line including inbound/outbound liquidity and alias"?

C-Otto commented on rebalance.py:
```diff
@@ -301,6 +318,14 @@ def get_argument_parser():
     )
+    #addition list all chanids
+    parser.add_argument(
+       "-la",
```
Why "la"?

C-Otto commented on rebalance.py:
```diff
@@ -301,6 +318,14 @@ def get_argument_parser():
+    #addition list all chanids
+    parser.add_argument(
+       "-la",
+        "--list-chanids",
```
Again, the list contains more than the channel IDs. Maybe "--compact-list"?

C-Otto commented on rebalance.py:
```diff
@@ -176,13 +176,28 @@ def show_channel(self, channel, reverse=False):
+        candidates = sorted(
+            self.lnd.get_channels(active_only=True),
+            key=lambda c: self.get_sort_key(c),
+            reverse=reverse
```
You can just use `reverse=False` here and remove the parameter.

C-Otto commented on rebalance.py:
```diff
@@ -176,13 +176,28 @@ def show_channel(self, channel, reverse=False):
+            reverse=reverse
+            )
+        for candidate in candidates:
+            print(f"ChanID: {format_boring_string(candidate.chan_id)} | {format_amount_green(get_local_available(candidate), 10)}|{format_amount(get_remote_available(candidate), 10)} | Alias: {format_alias(self.lnd.get_node_alias(candidate.remote_pubkey))}")
```
Please use 11 instead of 10 so that large channels aren't cramped.
I also think the output is still readable if you remove the "ChanID: " and "Alias: " prefixes.

C-Otto reviewed (changes requested):
Thanks a lot! Please have a look at my remarks.

blckbx commented:
- removed all comments 
- renamed parameter to "-cl" / "--compact-list" and changed description as suggested
- reformatted list and removed prefixes as suggested
- renamed method and removed parameter
-
